src/modifier/modifier-type.ts

```diff
@@ -1328,6 +1399,63 @@ export class FusePokemonModifierType extends PokemonModifierType {
   }
 }
 
+class AttackTypeBoosterRewardGenerator extends ModifierTypeGenerator {
+  constructor() {
+    super((party: Pokemon[], pregenArgs?: any[]) => {
+      if (pregenArgs && pregenArgs.length === 1 && pregenArgs[0] in PokemonType) {
+        return new AttackTypeBoosterReward(pregenArgs[0] as PokemonType, TYPE_BOOST_ITEM_BOOST_PERCENT);
+      }
+
+      const attackMoveTypes = party.flatMap(p =>
+        p
+          .getMoveset()
+          .map(m => m.getMove())
+          .filter(m => m instanceof AttackMove)
+          .map(m => m.type),
+      );
+      if (!attackMoveTypes.length) {
+        return null;
+      }
+
+      const attackMoveTypeWeights = new Map<PokemonType, number>();
+      let totalWeight = 0;
+      for (const t of attackMoveTypes) {
+        if (attackMoveTypeWeights.has(t)) {
+          if (attackMoveTypeWeights.get(t)! < 3) {
+            // attackMoveTypeWeights.has(t) was checked before
+            attackMoveTypeWeights.set(t, attackMoveTypeWeights.get(t)! + 1);
+          } else {
+            continue;
+          }
+        } else {
+          attackMoveTypeWeights.set(t, 1);
+        }
+        totalWeight++;
```

Comment:
```suggestion
      for (const t of attackMoveTypes) {
        const weight = attackMoveTypeWeights.get(t) ?? 0;
        if (weight < 3) {
          attackMoveTypeWeights.set(t, weight + 1)
          totalWeight++
        }
```

Shorter way of writing the same thing.